Holovibes/includes/data_struct/batch_input_queue.hh

```diff
@@ -49,23 +49,31 @@ class BatchInputQueue final : public DisplayQueue
 
     ~BatchInputQueue();
 
-    /*! \brief Enqueue a frame in the queue
+    /*! \brief Enqueues frames into the batch input queue with CUDA memory transfer.
      *
-     * Called by the producer.
-     * The producer is in the critical while enqueueing in a batch
-     * and exit this critical section when a batch of frames is full
-     * in order to let the resize occure if needed.
+     * This function handles the transfer of frames from the input source to the internal
+     * batch-based queue. Depending on the device type (CPU or GPU) and the memory copy
+     * direction, it ensures data consistency and synchronizes correctly with concurrent
+     * operations. Frames are enqueued in batches, and when a batch is full, the queue
+     * updates its state, potentially overwriting older batches if the queue is full.
+     *
+     * \param[in] frames Pointer to the source frames to be enqueued.
+     * \param[in] memcpy_kind Type of memory copy operation (e.g., cudaMemcpyHostToDevice).
+     * \param[in] nb_frame Number of frames to enqueue.
+     *
+     * \throws std::runtime_error If the memory copy direction is incompatible with the
+     *         current device type (e.g., copying to GPU while the queue is on CPU).
      */
-    void enqueue(const void* const input_frame, const cudaMemcpyKind memcpy_kind = cudaMemcpyDeviceToDevice);
-
-    // bool enqueue(void* elt, const cudaStream_t stream, const cudaMemcpyKind cuda_kind = cudaMemcpyDeviceToDevice);
+    void enqueue(const void* const frames,
+                 const cudaMemcpyKind memcpy_kind = cudaMemcpyDeviceToDevice,
+                 const int nb_frame = 1);
 
     /*! \brief Copy multiple
      *
      * Called by the consumer.
      *
-     * \param dest The destination queue
-     * \param nb_elts Number of elts to copy multiple (must be lower than batch_size)
+     * \param[in] dest The destination queue
+     * \param[in] nb_elts Number of elts to copy multiple (must be lower than batch_size)
```

Comment:
`cuda_kind` param is missing and dest should be param[out] if it is used as output